btc_parser/sources/output.move

```diff
@@ -27,3 +53,73 @@ public fun amount(output: &Output): u64 {
 public fun script_pubkey(output: &Output): vector<u8> {
     output.script_pubkey
 }
+
+public fun is_P2PHK(output: &Output): bool {
+    let script = output.script_pubkey();
+
+    script.length() == 25 &&
+		script[0] == OP_DUP &&
+		script[1] == OP_HASH160 &&
+		script[2] == OP_DATA_20 &&
+		script[23] == OP_EQUALVERIFY &&
+		script[24] == OP_CHECKSIG
+}
+
+public fun is_op_return(output: &Output): bool {
+    let script = output.script_pubkey;
+    script.length() > 0 && script[0] == OP_RETURN
+}
+
+public fun is_P2WPHK(output: &Output): bool {
+    let script = output.script_pubkey;
+    script.length() == 22 &&
+        script[0] == OP_0 &&
+        script[1] == OP_DATA_20
+}
+
+// TODO: add support script addresses.
+// TODO: check and verify the address to make sure we support it. Return error otherwise
+/// extract pkh from the output in P2PHK or P2WPKH
+/// return empty if the cannot extract public key hash
+public fun extract_public_key_hash(output: &Output): vector<u8> {
+    let script = output.script_pubkey;
+    if (output.is_P2PHK()) {
+        return vector_slice(&script, 3, 23)
+    } else if (output.is_P2WPHK()) {
+        return vector_slice(&script, 2, 22)
+    };
+    vector[]
+}
+
+/// Extracts the data payload from an OP_RETURN output in a transaction.
+/// script = OP_RETURN <data>.
+/// If transaction mined to BTC, then this must pass basic conditions
+/// include the conditions for OP_RETURN script.
+/// This why we only return the message without check size message.
+public fun op_return(output: &Output): vector<u8> {
+    let script = output.script_pubkey;
+
+    if (script.length() == 1) {
```

Comment:
```suggestion
    if (script.length() <= 1) {
```

Reply:
I think this will be good if we return option<vector<u8>>in this case. 
I will create task for this.